src/SavingsDai.sol

```diff
@@ -293,9 +297,13 @@ contract SavingsDai {
     }
 
     function deposit(uint256 assets, address receiver) external returns (uint256 shares) {
+        shares = deposit(assets, receiver, 0);
```

Comment:
I think there was a preference for using a shared internal function and making interfaces symmetric and external, see https://github.com/makerdao/dss-vest/blob/master/src/DssVest.sol#L223.
This is relevant also for `mint()`.

Reply:
Done

Reply:
Mmm we now have two internal `_mint` functions.
We should probably try to avoinf that.

Can maybe rename the lower level `_mint` to `_doMint`?
Then we'll have:
`mint`      => `_mint`        => `_doMint`
`deposit` => `_deposit`   => `_doMint`

Reply:
Done for `_doMint` as well as `_doBurn` for consistency. Also applied the styling to `permit()`.